common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java

```diff
@@ -68,6 +72,126 @@ public String regexSub(@LuaNotNil String str, @LuaNotNil String pattern, @LuaNot
         return str.replaceAll(pattern, replacement);
     }
 
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.debug_to_log"
+    )
+    public void debugToLog(@LuaNotNil String data) {
+        if (Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) {
+            GoofyPlugin.LOGGER.debug("[" + owner.entityName + "] - " + data);
+        }
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.info_to_log"
+    )
+    public void infoToLog(@LuaNotNil String data) {
+        if (Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) {
+            GoofyPlugin.LOGGER.info("[" + owner.entityName + "] - " + data);
+        }
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.warn_to_log"
+    )
+    public void warnToLog(@LuaNotNil String data) {
+        if (Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) {
+            GoofyPlugin.LOGGER.warn("[" + owner.entityName + "] - " + data);
+        }
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.error_to_log"
+    )
+    public void errorToLog(@LuaNotNil String data) {
+        if (Configs.LOG_OTHERS.value || FiguraMod.isLocal(owner.owner)) {
+            GoofyPlugin.LOGGER.error("[" + owner.entityName + "] - " + data);
+        }
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"data"}
+            )
+        },
+        value = "goofy.trace_to_log"
+    )
+    public void traceToLog(@LuaNotNil String data) {
```

Comment:
tracing seems like it wouldn't be useful, as it outputs the *Java* trace

Reply:
I added it to be complete with all supported options

Reply:
So I personally want to leave it in. Didn't merge yet I wanna change one thing

Reply:
Alright makes sense